src/D2L.CodeStyle.TestAnalyzers/Extensions/Microsoft.CodeAnalysis.cs

```diff
@@ -26,6 +26,19 @@ public static class RoslynExtensions {
 			return false;
 		}
 
+		public static bool IsErrorType( this ITypeSymbol symbol ) {

```

Comment:
In some places I lifted the null check out because it was convenient.
For ISymbol I just replaced any problematic usages with the inline checks. I kept this one because there is an extra check (Kind and TypeKind).

I don't really like these overloads much, it seems pretty imprecise, but I didn't want to look too closely to see if all the cases were necessary.